dandiapi/api/services/dandiset/__init__.py

```diff
@@ -20,7 +20,7 @@ def create_dandiset(
     *,
     user,
     identifier: int | None = None,
-    embargo: bool,
+    embargo_config: dict,
```

Comment:
Making this an opaque `dict` significantly decreases the readability of this code, I think we should just pull the dict contents into separate function arguments -


```suggestion
    embargo: bool,
    funding_source: str | None,
    award_number: str | None,
```

However, I don't think this is that important because I think we have a bigger problem here - the `create_dandiset` function has become bloated due to the increasing divergence between open/embargoed logic. I propose that we break this out into two separate functions (like we [already do](https://github.com/dandi/dandi-archive/blob/a85249e45b3f005ba3d7596cba091e16e9a4d265/web/src/rest.ts#L213-L243) for the equivalent functions in the frontend). See #2507 for my suggested change here

Reply:
I was trying to implement the feature with minimal changes. I'd be fine with refactoring to expose these args and to create two separate entry points for creating public and embargoed dandisets